docs/source/java/flight.rst

```diff
@@ -0,0 +1,216 @@
+expose a cancel method which allows terminating a call early.
+
+.. code-block:: Java
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+
+        Iterator<Result> resultIterator = tutorialFlightClient.doAction(
+                new Action("test-timeout"),
+                CallOptions.timeout(2, TimeUnit.SECONDS)
+        );
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+On the server side, timeouts are transparent. For cancellation, the server needs to manually poll
+`setOnCancelHandler` or `isCancelled` to check if the client has cancelled the call, and if so,
+break out of any processing the server is currently doing.
+
+.. code-block:: Java
+
+    // Client
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+        try(FlightStream flightStream = flightClient.getStream(new Ticket(new byte[]{}))) {
+            // ...
+            flightStream.cancel("tutorial-cancel", new Exception("Testing cancellation opion!"));
+        }
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+    // Server
+    @Override
+    public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+        // ...
+        listener.setOnCancelHandler(()->{
+                    // Implement logic to handle cancellation option
+                });
+    }
+
+Enabling TLS
+============
+
+TLS can be enabled when setting up a server by providing a
+certificate and key pair to `FlightServer.builder.useTls`.
+
+On the client side, use `FlightClient.builder.trustedCertificates`.
+
+Enabling Authentication
+=======================
+
+.. warning:: Authentication is insecure without enabling TLS.
+
+Handshake-based authentication can be enabled by implementing
+`ServerAuthHandler`. Authentication consists of two parts: on
+initial client connection, the server and client authentication
+implementations can perform any negotiation needed; then, on each RPC
+thereafter, the client provides a token. The server authentication
+handler validates the token and provides the identity of the
+client. This identity can be obtained from the
+`CallContext.peerIdentity`.
+
+.. code-block:: Java
+
+    // Client
+    FlightClient client = FlightClient.builder().build();
+    client.authenticateBasic("user", "password");
+    // Server
+    FlightServer server = FlightServer.builder().authHandler(new BasicServerAuthHandler(validator)).build();
+    // CallHeaders printed on the Flight Server:
+    Metadata(content-type=application/grpc,user-agent=grpc-java-netty/1.44.1,auth-token-bin=bXlfdG9rZW4,grpc-accept-encoding=gzip)
```

Comment:
I'd rather we omit this in favor of adding a full example to the Cookbook. Also because the server doesn't print that by default, *and* this is a different authentication method than what we just described.

Reply:
Deleted